src/Microsoft.Extensions.Configuration.AzureAppConfiguration/DomainVerifier.cs

```diff
@@ -0,0 +1,145 @@
+﻿// Copyright(c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Net.Security;
+using System.Net.Sockets;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration
+{
+    /// <summary>
+    /// Verify the domain name of the endpoint matches the certificate.
+    /// </summary>
+    internal static class DomainVerifier
+    {
+        private const int TlsPort = 443;
+        private const string SubjectAltNameOid = "2.5.29.17";
+        private const string CommonName = "CN=";
+        private const string MultiDomainWildcard = "*.";
+
+        public static async Task<string> GetValidDomain(Uri originEndpoint, string srvHostName)

```

Comment:
This should be able to return multiple valid domains if there are multiple matches.

Reply:
Make sense, updated